src/Workspaces/Core/Portable/Classification/IRemoteSemanticClassificationCacheService.cs

```diff
@@ -27,7 +27,7 @@ internal interface IRemoteSemanticClassificationCacheService
         /// </summary>
         /// <param name="checksum">Pass in <see cref="DocumentStateChecksums.Text"/>.  This will ensure that the cached
         /// classifications are only returned if they match the content the file currently has.</param>
-        Task<SerializableClassifiedSpans?> GetCachedSemanticClassificationsAsync(
+        ValueTask<SerializableClassifiedSpans?> GetCachedSemanticClassificationsAsync(
```

Comment:
what's the intution on Task/ValueTask in the context of OOP/remoting.  why would i pick one or the other in the future?

Reply:
IT's just a question of what the TryInvoke helpers use in the delegates they take. I opted for ValueTask since it's not gonna allocate unnecessarily. That choice forces all other signatures unless we convert between ValueTask and Task.